azext_iot/digitaltwins/providers/connection/builders.py

```diff
@@ -0,0 +1,285 @@
+            raise CLIError("{} retrieve Event Hub Consumer Group.".format(ERROR_PREFIX))
+
+        self.eh_namespace_resource_id = (
+            "/subscriptions/{}/resourceGroups/{}/providers/Microsoft.EventHub/namespaces/{}".format(
+                eh_subscription,
+                eh_resource_group,
+                eh_namespace,
+            )
+        )
+        logger.debug(FINISHED_CHECK_RESOURCE_LOG_MSG.format("Event Hub"))
+
+        self.add_dt_role_assignment(
+            role="Azure Event Hubs Data Owner",
+            resource_id=f"{self.eh_namespace_resource_id}/eventhubs/{eh_entity_path}"
+        )
+
+    def validate_adx(
+        self,
+        adx_cluster_name: str,
+        adx_database_name: str,
+        adx_resource_group: str,
+        adx_subscription: str,
+    ):
+        api_version = "api-version=2021-01-01"
+        self.adx_resource_id = (
+            "/subscriptions/{}/resourceGroups/{}/providers/Microsoft.Kusto/clusters/{}".format(
+                adx_subscription,
+                adx_resource_group,
+                adx_cluster_name
+            )
+        )
+
+        adx_cluster_op = self.cli.invoke(
+            "rest --method get --url {}?{}".format(
+                self.adx_resource_id,
+                api_version
+            )
+        )
+        if not adx_cluster_op.success():
+            raise CLIError("{} retrieve Cluster.".format(ERROR_PREFIX))
+
+        adx_cluster_meta = adx_cluster_op.as_json()
+        self.adx_cluster_uri = adx_cluster_meta["properties"]["uri"]
+        self.adx_location = adx_cluster_meta["location"].lower().replace(" ", "")
+
+        adx_database_op = self.cli.invoke(
+            "rest --method get --url {}/databases/{}?{}".format(
+                self.adx_resource_id,
+                adx_database_name,
+                api_version
+            )
+        )
+        if not adx_database_op.success():
+            raise CLIError("{} retrieve Database.".format(ERROR_PREFIX))
+
+        logger.debug(FINISHED_CHECK_RESOURCE_LOG_MSG.format("Azure Data Explorer"))
+
+        self.add_dt_role_assignment(
+            role="Contributor",
+            resource_id=f"{self.adx_resource_id}/databases/{adx_database_name}"
+        )
+        self.add_adx_principal(adx_database_name, api_version)
+
+    def add_dt_role_assignment(self, role, resource_id):
+        role_str = RBAC_ROLE_MSG.format(role, resource_id)
+        logger.debug(TRY_ADD_ROLE_LOG_MSG.format(role_str))
+        if not (self.yes or prompt_y_n(msg=ADD_INPUT_MSG.format(role_str), default="y")):
+            print(SKIP_MSG.format(role_str))
+            return
+
+        current_roles_op = self.cli.invoke(
+            "role assignment list --role '{}' --assignee {} --scope {}".format(
+                role,
+                self.dt.identity.principal_id,
+                resource_id
+            )
+        )
+
+        if current_roles_op.success() and len(current_roles_op.as_json()) > 1:
```

Comment:
For the second condition is it supposed to be `>= 1`?

Reply:
assuming that the rbac api isnt wonky, there should be only 0 items or 1 item returned from the call since it is looking for the specific resource, scope and role. There isn't any include inherited so I am not sure where a duplicate can come from.

But I can change it so we would be on the safe side (as do not trust the rbac api)

Reply:
Discussed offline.